### PR TITLE
[#570] Add OpenShift S2I deployment

### PR DESCRIPTION
**Pull request by ctron**

### CODE REVIEW

File: site/content/deployment/openshift_s2i.md [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+All examples in the following sub-sections assume that you are located in the
+`examples` directory.
+
+### Extract certificates
+
+In order to connect the external consumer to EnMasse, we need to use a
+proper SSL certificate. We can extract one from the OpenShift using the
+following command:
+
+    oc extract secret/external-certs-messaging --to=target/keys -n enmasse
```
Comment (dejanb) [inline site/content/deployment/openshift_s2i.md]: target/keys doesn't work by default ... maybe using target/config/hono-demo-certs-jar/ as we do in other document is a better option

File: site/content/deployment/openshift_s2i.md [hunk 2]
```diff
@@ -34,6 +34,21 @@ the version of your OpenShift cluster. This guide was tested with
 OpenShift 3.7.2. It should work with older or newer versions as well, but that
 is untested.
 
+{{% warning title="Curl on Mac OS X" %}}
+The `curl` binary on Mac OS X suffers from an issue with TLS SNI. Also see: https://github.com/curl/curl/issues/1533
+
+As the use of SNI is required for Kubernetes/OpenShift when it comes to routing
```
Comment (sophokles73) [inline site/content/deployment/openshift_s2i.md]: My understanding is that this is just an issue on OpenShift when using routes (which do not exist on Kubernetes, do they?)

Comment (ctron) [reply site/content/deployment/openshift_s2i.md]: This effects all cases where TLS SNI is used. In the issue mentioned I saw some references to kubernetes as well. As I don't have any hands on experience with plain kubernetes, I don't know. I can remove it, if you think it is better.

File: site/content/deployment/openshift_s2i.md (hunk 1 above)
Comment (ctron) [reply site/content/deployment/openshift_s2i.md]: Well the following commands all have the changed paths. Which makes the commands a bit shorter. But I agree, I you want to re-use the commands from the other tutorials, you might run into issued. I will sync this with the other examples.

**Pull request merged**